Stage 1 frontend and api

ValTsan opened the pull request.

ValTsan committed:
Add readme.md

ValTsan committed:
Initial commit

ValTsan committed:
Add icons and photos

ValTsan committed:
Add initial project structure

ValTsan committed:
Added footer and fonts

ValTsan committed:
Added header and main

ValTsan committed:
Add LoginModal

ValTsan committed:
Add AboutMe

ValTsan committed:
Add footer

ValTsan committed:
Add header and css

ValTsan committed:
Fix icons for footer

ValTsan committed:
Fix header css

ValTsan committed:
Added ModalWithForm and css

ValTsan committed:
Add registerModal and css

ValTsan committed:
Add savedNewsHeader and css

ValTsan committed:
Fix news header

ValTsan committed:
Fix news header

ValTsan committed:
Add news api

ValTsan committed:
Add search form section and results section

ValTsan committed:
Add preloader

ValTsan committed:
Fix css

ValTsan committed:
Fix main css

ValTsan committed:
Add login modal and css

ValTsan committed:
Update/add saved news and css

ValTsan committed:
Fix auth.js

ValTsan committed:
Fix fonts imports

ValTsan committed:
Fix css for search form

ValTsan committed:
Add news card and css

ValTsan committed:
Fix/add css

ValTsan committed:
Fix/change add css for results

ValTsan committed:
Fix/change search form css

ValTsan committed:
Fix/align items on search form

ValTsan committed:
Fix/change css for about section

ValTsan committed:
Fix/change css for about me and header nav

ValTsan committed:
Fix/change css for media

ValTsan committed:
Fix results css and logic

ValTsan committed:
Fix results logic and css

ValTsan committed:
Fix header css and media css

ValTsan committed:
Fix header log media 480

ValTsan committed:
Refactor css for header main screen

ValTsan committed:
Fix header overflow

ValTsan committed:
Fix results css and cards

ValTsan committed:
Fix css for news card and media

ValTsan committed:
Fix modal css and media

ValTsan committed:
Add form validation on register modal

ValTsan committed:
Add validation for login modal

ValTsan committed:
Clean up css and app

ValTsan committed:
Fix news card css

ValTsan committed:
Add git pages

ValTsan committed:
Add git pages

ValTsan committed:
Add git pages

ValTsan committed:
Add link to readme

ValTsan committed:
Git pages deployment fix

ValTsan committed:
Fix API key

ValTsan committed:
Fix css for news sign in bookmark

ValTsan committed:
Add api

ValTsan committed:
Fix header while logged in

ValTsan committed:
Fix profile pic import

ValTsan committed:
Fix profile pic path

ValTsan committed:
Add keywords

ValTsan committed:
Fix keyword box

ValTsan committed:
Fix keyword

ValTsan committed:
Fix saved news card

ValTsan committed:
fix news source css

ValTsan committed:
Fix css for span source

ValTsan committed:
Fix rendering duplicated on saved news

ValTsan committed:
Console logs clean up

n817 commented:
Hello! Great job with your first diploma stage! The project looks very good in general, but will require some refactoring.

# Summary:
### Performance Criteria

- [ ] <!--12.8--> All the blocks from the brief have been coded. The layout matches the designs for all resolutions from the mockup. Differences in size should not visually contradict the layout or get out of the grid in the layout.

* A font is not valid: `Roboto` and `Roboto-slab` fonts are not rendered on the page.

* On desktop 1440px:
    - The `modal` block doesn't match the design, it's necessary to fix it.

* On mobile 320px:
    - Incorrect width of `news-card` elements. The project should match the design in the Figma.
    - The `footer` block doesn't match the design, it's necessary to fix it.

You can use PerfectPixel plugin to compare your project with the design: https://www.welldonecode.com/perfectpixel/

- [ ] <!--12.6--> The project is adapted for different screen resolutions and looks correct on all intermediate resolutions.  There is no horizontal scrolling on resolutions of 320px and up. The scrollbar can't be hidden using the overflow: hidden declaration.

* The page scrolls horizontally on 320px-350px screen resolutions. It's necessary to fix it.

To find the elements that cause horizontal scrolling, you can use the `outline` CSS property and implement it globally:

`* {
  outline: 1px solid red;
}`

All the elements on the page will be outlined with a red line, and you'll be able to quickly find which of them is the cause of scrolling. Don't forget to delete this code before submitting it for review.

- [x] <!--12.6--> Navigation between pages and links to external resources are working correctly: there are no broken links or links leading to an empty anchor on the page, and external links open in a new tab. All project pages can be accessed, and hidden blocks can be displayed.
- [x] <!--4.2--> Classes are named according to BEM specifications.
- [ ] <!--4.2--> Semantic HTML is used, meaning that semantic tags are used. All elements are used correctly (e.g., a paragraph must be a paragraph, a list must be a list). The DOM tree structure doesn't consist only of `<div />` containers.

* For elements containing text, use `h1`-`h6` tags (for headers) and `p` (for paragraphs and subheadings) instead of `<div>`.
Please fix the tag for the `header__logo` element.

- [x] <!--4.2--> The correct approach for positioning elements has been chosen and is described using the correct syntax. For example, when positioning an element absolutely, its parent block is relatively positioned.
- [x] <!--4.2--> `flex` or `grid` layouts are used to arrange elements whenever applicable.
- [x] <!--4.2--> Infrastructural project files are created using CRA.
- [x] <!--4.2--> The popup windows have opening and closing functionality implemented.
- The markup has been converted into JSX:
  - [x] <!--4.2--> The markup is included within parentheses `( )`.
  - [x] <!--4.2--> The markup has been moved into the corresponding components.
- [x] <!--4.2--> Your project contains:
  - [x] An `images`
  - [x] A `components` folder with JS and CSS files for the components.
  - [x] A `fonts` folder.
- [x] <!--4.2--> There are no linting errors or warnings.

### Best Practices

- [ ] <!--2.16--> Buttons, input fields, and links are implemented in all of the states specified in the design.

* All interactive elements (buttons and links) should have a `:hover` state, according to the design.

- [x] <!--2.14--> The design reuses components wherever possible.
- [ ] <!--2.14--> Fonts are connected using `@font-face`.

* A font is not valid: `Roboto` and `Roboto-slab` fonts are not rendered on the page.

- [x] <!--2.14--> The icons are exported in SVG format.
- [x] <!--2.14--> Form elements should be highlighted when focused.
- [x] <!--2.14--> The form must have placeholders, and there are required properties for the fields.
- [x] <!--2.14--> The use of `reset.css` is prohibited.


### Recommendations

- [x] <!--1.67--> System fonts are connected as alternatives to each of your fonts.
- [x] <!--1.67--> Images have an `alt` attribute containing appropriate values.
- [ ] <!--1.66--> Raster and vector images have been optimized.

* It's better to reduce size of `ProfilePic.jpeg` image. Use the optimized images to save bandwidth and faster loading time of your website. You can use online-compression service, e.g. https://tinypng.com

## Number of points: 64

ValTsan committed:
Fix fonts and modal

ValTsan committed:
Fix 320 media news card and about

ValTsan committed:
Fix footer layout on mobile view

ValTsan committed:
Fix tags for header logo

ValTsan committed:
Fix profile pic

ValTsan commented:
All required changes have been implemented:
	•	Fixed custom font rendering (Roboto & Roboto Slab)
	•	Updated modal styling to match the 1440px design using PerfectPixel
	•	Corrected media card widths for 320px screens
	•	Footer layout is now responsive and consistent across all breakpoints
	•	Replaced non-semantic tags with semantic elements for the header logo
	•	Added hover states for all interactive elements (buttons, links, inputs)

No horizontal scrolling at 320px and up. Semantic HTML and accessibility standards have been applied throughout.

Let me know if anything else needs adjustment.

ImaliaNsk commented:
Thanks for the corrections, the project has been improved significantly, and it looks more like in the design. However, a few issues still need your attention, they are mostly small and quick to fix, I believe you can handle them with ease! Good luck!

# Summary:
### Performance Criteria

- [ ] <!--12.8--> All the blocks from the brief have been coded. The layout matches the designs for all resolutions from the mockup. Differences in size should not visually contradict the layout or get out of the grid in the layout.

* On mobile 320px: the cards are too narrow https://prnt.sc/Z5tHP2I2FCQF (both pages)
* https://prnt.sc/rYuIHaQt2S_7 https://prnt.sc/DLd1lrDxBAgR https://prnt.sc/LKM1KR4YxXmh https://prnt.sc/rrnwwDP85nEd Incorrect font properties. It's necessary to check font-family, font-size, line-height
* https://prnt.sc/sPv0eAPIjjUt The text of the link shouldn't be underlined
* The source text inside cards should be aligned to the bottom: https://prnt.sc/1Ohs-MCmP5-9
* https://prnt.sc/87sO4t3savfA When the search is active, "Search results" title shouldn't be dispayed, and the preloader shouldn't be cropped, in fact, we need a bottom padding in this block, some space between the element and the bottom border of the block.
* https://prnt.sc/fI_D4WLj0mhj Navigation menu looks incorrectly

- [ ] <!--12.6--> The project is adapted for different screen resolutions and looks correct on all intermediate resolutions.  There is no horizontal scrolling on resolutions of 320px and up. The scrollbar can't be hidden using the overflow: hidden declaration.

* Home page scrolls on 320 - 351, 481 - 503 https://prnt.sc/ywXQsYXd5U4t
* Text in "about" shouldn't get that narrow: https://prnt.sc/f3QoJfjKEyvw https://prnt.sc/-4Z8abHXYTMl

- [x] <!--12.6--> Navigation between pages and links to external resources are working correctly: there are no broken links or links leading to an empty anchor on the page, and external links open in a new tab. All project pages can be accessed, and hidden blocks can be displayed.
- [x] <!--4.2--> Classes are named according to BEM specifications.
- [ ] <!--4.2--> Semantic HTML is used, meaning that semantic tags are used. All elements are used correctly (e.g., a paragraph must be a paragraph, a list must be a list). The DOM tree structure doesn't consist only of `<div />` containers.

* Each page needs only one h1 heading (so `header__logo` shouldn't be h1, best to make it an <a>, it can also be `<p>` or another level of h)

- [x] <!--4.2--> The correct approach for positioning elements has been chosen and is described using the correct syntax. For example, when positioning an element absolutely, its parent block is relatively positioned.
- [x] <!--4.2--> `flex` or `grid` layouts are used to arrange elements whenever applicable.
- [x] <!--4.2--> Infrastructural project files are created using CRA.
- [x] <!--4.2--> The popup windows have opening and closing functionality implemented.
- The markup has been converted into JSX:
  - [x] <!--4.2--> The markup is included within parentheses `( )`.
  - [x] <!--4.2--> The markup has been moved into the corresponding components.
- [x] <!--4.2--> Your project contains:
  - [x] An `images`
  - [x] A `components` folder with JS and CSS files for the components.
  - [x] A `fonts` folder.
- [x] <!--4.2--> There are no linting errors or warnings.

### Best Practices

- [ ] <!--2.16--> Buttons, input fields, and links are implemented in all of the states specified in the design.

* Hover effect on a card bookmark from the mockup: https://prnt.sc/o7v7VfcnRLh8
* Hover effect on trash can icon from the mockup: https://prnt.sc/KFTCWRfN-fBQ
* "Search" button is missing an active state
* "Show more" button hover effect doesn't match the design.

- [x] <!--2.14--> The design reuses components wherever possible.
- [x] <!--2.14--> Fonts are connected using `@font-face`.
- [x] <!--2.14--> The icons are exported in SVG format.
- [x] <!--2.14--> Form elements should be highlighted when focused.
- [x] <!--2.14--> The form must have placeholders,  and there are required properties for the fields.
- [x] <!--2.14--> The use of `reset.css` is prohibited.


### Recommendations

- [ ] <!--1.67--> System fonts are connected as alternatives to each of your fonts.

* Please fix all similar cases: https://prnt.sc/7BGzyBNHNx52

- [x] <!--1.67--> Images have an `alt` attribute containing appropriate values.
- [ ] <!--1.66--> Raster and vector images have been optimized.

* Main background image shouldn't be in svg, please use raster graphics format for pictures like this (jpg or png).

## Number of points: 65

ValTsan committed:
Fix news cards on 320 px and 480px

ValTsan committed:
Fix source news card

ValTsan committed:
Fix preloader margin bottom

ValTsan committed:
Fix mobile menu background

ValTsan committed:
Fix nav bar and media css

ValTsan committed:
Fixx fonts and layout for media

ValTsan committed:
Fix and optimize header background image

ValTsan committed:
Fix preloader margin

ImaliaNsk commented:
Congratulations, the project is accepted! I wish you success on the next stages and in your future projects!

# Summary:
### Performance Criteria

- [ ] <!--12.8--> All the blocks from the brief have been coded. The layout matches the designs for all resolutions from the mockup. Differences in size should not visually contradict the layout or get out of the grid in the layout.

* https://prnt.sc/WYNW6WLjfitW Please check the font-size and alignment of the text
* https://prnt.sc/mzJ16MSYKWWO https://prnt.sc/qjiDBGZI_uVS All content sections need the same side gaps so that the content looks aligned. Now only the cards section gaps match the design, they are bigger for header and "about", and smaller in the footer.
* https://prnt.sc/LKM1KR4YxXmh Incorrect font properties. It's necessary to check font-family, font-size, line-height
* https://prnt.sc/m11POWIRdYg3 Elements in the navigation menu should be aligned, please refer to the figma for correct margin/gap values.
* https://prnt.sc/AU1ZgumZh_a7 Link text shouldn't be underlined. Also, the border line at the bottom of the header is missing.

- [x] <!--12.6--> The project is adapted for different screen resolutions and looks correct on all intermediate resolutions.  There is no horizontal scrolling on resolutions of 320px and up. The scrollbar can't be hidden using the overflow: hidden declaration.
- [x] <!--12.6--> Navigation between pages and links to external resources are working correctly: there are no broken links or links leading to an empty anchor on the page, and external links open in a new tab. All project pages can be accessed, and hidden blocks can be displayed.
- [x] <!--4.2--> Classes are named according to BEM specifications.
- [x] <!--4.2--> Semantic HTML is used, meaning that semantic tags are used. All elements are used correctly (e.g., a paragraph must be a paragraph, a list must be a list). The DOM tree structure doesn't consist only of `<div />` containers.
- [x] <!--4.2--> The correct approach for positioning elements has been chosen and is described using the correct syntax. For example, when positioning an element absolutely, its parent block is relatively positioned.
- [x] <!--4.2--> `flex` or `grid` layouts are used to arrange elements whenever applicable.
- [x] <!--4.2--> Infrastructural project files are created using CRA.
- [x] <!--4.2--> The popup windows have opening and closing functionality implemented.
- The markup has been converted into JSX:
  - [x] <!--4.2--> The markup is included within parentheses `( )`.
  - [x] <!--4.2--> The markup has been moved into the corresponding components.
- [x] <!--4.2--> Your project contains:
  - [x] An `images`
  - [x] A `components` folder with JS and CSS files for the components.
  - [x] A `fonts` folder.
- [x] <!--4.2--> There are no linting errors or warnings.

### Best Practices

- [ ] <!--2.16--> Buttons, input fields, and links are implemented in all of the states specified in the design.

* Hover effect on a card bookmark from the mockup: https://prnt.sc/o7v7VfcnRLh8
* Hover effect on trash can icon from the mockup: https://prnt.sc/KFTCWRfN-fBQ

- [x] <!--2.14--> The design reuses components wherever possible.
- [x] <!--2.14--> Fonts are connected using `@font-face`.
- [x] <!--2.14--> The icons are exported in SVG format.
- [x] <!--2.14--> Form elements should be highlighted when focused.
- [x] <!--2.14--> The form must have placeholders,  and there are required properties for the fields.
- [x] <!--2.14--> The use of `reset.css` is prohibited.


### Recommendations

- [x] <!--1.67--> System fonts are connected as alternatives to each of your fonts.
- [x] <!--1.67--> Images have an `alt` attribute containing appropriate values.
- [ ] <!--1.66--> Raster and vector images have been optimized.

* It's better to reduce the size of images. Using optimized images will save bandwidth and result in faster website loading times.

## Number of points: 83
